### PR TITLE
set and transform nodeID

### PR DESCRIPTION
**Pull request by kinarashah**

### CODE REVIEW

**Comment by ibuildthecloud** [comment]
What does this do?

**Comment by kinarashah** [comment]
@ibuildthecloud on creating ingress with node scheduling, the node name was received as `clusterName:nodeName` so just setting it to nodeName on creation and transforming it back for frontend 
forgot to mention, #11709

**Comment by alena1108** [comment]
@kinarashah you've meant workload, right?

File: pkg/api/customization/workload/aggregate_store.go [hunk 1]
```diff
@@ -263,6 +268,31 @@ func setSecrets(apiContext *types.APIContext, data map[string]interface{}) {
 	}
 }
 
+func setScheduling(apiContext *types.APIContext, data map[string]interface{}) {
+	if _, ok := data["scheduling"]; ok {
```
Comment (ibuildthecloud) [inline pkg/api/customization/workload/aggregate_store.go]: There's an API `values.GetValue(map, "key", "key", "key")` that may help here.  There also a corresponding `SetValue` method.

**Comment by alena1108** [comment]
@kinarashah I've also noticed that we have scheduling mapper for pod spec, wonder if the fix should really go there inhttps://github.com/rancher/types/blob/master/mapper/scheduling.go

File: pkg/api/customization/workload/aggregate_store.go (hunk 1 above)
Comment (kinarashah) [reply pkg/api/customization/workload/aggregate_store.go]: Thank you, have updated using get and set value methods

**Comment by kinarashah** [comment]
yes @alena1108 sorry meant workload instead of ingress

File: pkg/api/customization/workload/aggregate_store.go [hunk 2]
```diff
@@ -265,6 +270,25 @@ func setSecrets(apiContext *types.APIContext, data map[string]interface{}) {
 	}
 }
 
+func setScheduling(apiContext *types.APIContext, data map[string]interface{}) {
+	if nodeID := convert.ToString(values.GetValueN(data, "scheduling", "node", "nodeId")); nodeID != "" {
+		nodeName := getNodeName(apiContext, nodeID)
```
Comment (alena1108) [inline pkg/api/customization/workload/aggregate_store.go]: should we skip if nodeName is ""?

File: pkg/api/customization/workload/aggregate_store.go [hunk 3]
```diff
@@ -338,3 +362,28 @@ func getDomain(image string) string {
...
+	if err != nil {
+		logrus.Errorf("failed to save state on workload: %v", data["id"])
+		return
+	}
+
+	values.PutValue(data, string(content), "annotations", "workload.cattle.io/state")
+}
+
+func getState(data map[string]interface{}) map[string]string {
+	state := make(map[string]string)
```
Comment (alena1108) [inline pkg/api/customization/workload/aggregate_store.go]: do state := map[string]string{}

**Pull request merged**